geode-core/src/main/java/org/apache/geode/internal/cache/entries/AbstractRegionEntry.java

```diff
@@ -2073,8 +2073,16 @@ private boolean processGatewayTag(EntryEvent cacheEvent) {
     if (tagTime == VersionTag.ILLEGAL_VERSION_TIMESTAMP) {
       return true; // no timestamp received from other system - just apply it
     }
-    if ((tagDsid == stampDsid && tagTime > stampTime) || stampDsid == -1) {
+    // According to GatewayConflictResolver's java doc, it will only be used on tag with different
+    // distributed system id than stamp's
+    if (stampDsid == -1) {
       return true;
+    } else if (tagDsid == stampDsid) {
+      if (tagTime >= stampTime) {
```

Comment:
Is there a reason why we are now checking >= instead of just >?

Reply:
This is the bug I fixed yesterday. In current code, we did not throw CME immediately when found tagDsid == stampDsid and tagTime <= stampTime. Instead, we throw CME when 
tagDsid == stampDsid and tagTime < stampTime. See following code:
    if (tagTime > stampTime || tagTime == stampTime && tagDsid >= stampDsid) {
      if (isDebugEnabled) {
        logger.debug("allowing event");
      }
      return true;
    }

I will split the test into 2, another for resolver.